smac/scenario/scenario.py

```diff
@@ -250,12 +255,24 @@ def extract_instance_specific(instance_list):
                     self.feature_fn)[1]
 
         if self.feature_dict:
-            self.n_features = len(
-                self.feature_dict[list(self.feature_dict.keys())[0]])
             self.feature_array = []
             for inst_ in self.train_insts:
                 self.feature_array.append(self.feature_dict[inst_])
             self.feature_array = numpy.array(self.feature_array)
+            
+            # reduce dimensionality of features of larger than PCA_DIM
+            if self.feature_array.shape[1] > self.PCA_DIM:
+                X = self.feature_array
+                # scale features
+                X = (X - X.min()) / (X.max() - X.min())
+                X = numpy.nan_to_num(X) # if features with max == min
+                #PCA
+                pca = PCA(n_components=self.PCA_DIM)
+                self.feature_array = pca.fit_transform(X)
+                self.n_features = self.PCA_DIM
```

Comment:
This will only be set if the number of features is higher than the number of PCA dimensions. What happens in the other case?